docs/reference/multi_site_capabilities.rst

```diff
@@ -0,0 +1,47 @@
+.. _multi_site_capabilities:
+
+Multi-Site Capabilities
+=======================
+
+Overview
+--------
+
+Wagtail provides multi-site capability via the :ref:`site-model-ref` model.  One or more sites can be created via the Sites option under the admin Settings menu.  A root page is chosen and a :attr:`~wagtail.core.models.Site.hostname` is defined for each site that is created.
+
+:ref:`settings` can be defined by creating a model that inherits from ``BaseSetting``.
+
+While Wagtail provides multi-site capability and adds the ``site`` object to the request, multi-tenant capability is not currently supported.  Therefore, if more than one site is set up, any user who has access to the admin can log in through any hostname that has been defined for a site.  After logging in, however, all user rights that were assigned via Groups are honored.
```

Comment:
This point requires a very specific definition of multi-tenancy. In my opinion, Wagtail's support for multi-tenancy is not inconsistent with the current Wikipedia definition:

> "A tenant is a group of users who share a common access with specific privileges to the software instance. With a multitenant architecture, a software application is designed to provide every tenant a dedicated share of the instance - including its data, configuration, user management, tenant individual functionality and non-functional properties."

https://en.wikipedia.org/wiki/Multitenancy

Reply:
I understood from @gasman in [#4288](https://github.com/wagtail/wagtail/issues/4288#issuecomment-365914981) that Wagtail is not completely multi-tenant yet.  IMHO groups/permissions is not the ideal way to achieve true multi-tenancy.  Serving HTML files is not currently handled by groups/permissions.  It seems that, until `host` is central to the logic of serving model data and assets (html files, images, documents), that true multi-tenancy is not achieved.  (As I'm new to Wagtail in the last 18 months, I make that statement not as a conclusion - I have lots to learn!  I'm simply thinking that multi-tenancy should work like one of the various Django multi-tenant packages work).  Perhaps we could discuss multi-tenancy at the upcoming conference in Philadelphia - not necessarily as a discussion topic but at least informally.

Reply:
> Serving HTML files is not currently handled by groups/permissions

If you mean serving Wagtail pages, then that's not the case; you can restrict sections of the tree (which may correspond to sites) to users in specific groups:

https://docs.wagtail.io/en/v2.5.1/advanced_topics/privacy.html

> Perhaps we could discuss multi-tenancy at the upcoming conference in Philadelphia - not necessarily as a discussion topic but at least informally

For sure! Meanwhile I'm interested in @gasman's opinion (and anyone else's).

Reply:
When I refer to HTML files, I'm actually referring to different versions of an HTML file being able to be used for a `Page` model.  I think that the easiest way to begin thinking about multi-tenancy is by thinking in terms of homogeneous tenants (I gave the example of Chambers of Commerce [here](https://github.com/wagtail/wagtail/issues/4288#issue-297240375)).  In this case each tenant would have basically the same information needs and so a single set of Wagtail `Page` definitions could be made to service all tenants.  However, each Chamber would want to have their own look, and so when the `template` for `HomePage(Page)` says `template = 'home.html'`, then you have to ask which Chamber's `home.html` file is to be used?  In a comment [here ](https://stackoverflow.com/questions/56449188/having-a-404-for-each-site-in-a-wagtail-multisite-setup/56526327#56526327) I stated:

> I've started using a themes folder. The themes folder is at the first level in the project and within it are folders that each hold all the assets for a particular theme. In my ideal world, you would associate a theme with a site. 

Of course `settings` have to be configured appropriately so that the templates and the static assets can be found.  Multi-tenant or not, it makes much more sense to me to have everything related to a specific site and its appearance (HTML, CSS, JS, and image files, where appropriate) in a single named theme folder within the `themes` folder.  HTML files such as `allauth` files COULD still be stored in a `templates` folder at the root level of the project, but I've even started creating an `allauth` folder within my named theme folder because invariably I have to do a little customization of the `allauth` templates too.  This serves to keep EVERYTHING related to a particular website together with the benefit that, if I ever want to share a theme, then everything is in a single folder and not scattered across several apps.

Reply:
This project hasn't been updated for a couple of years, but - as far as I understand - is designed to solve exactly this problem:

https://github.com/moorinteractive/wagtail-themes

It may need updating to work with recent Wagtail versions, but that could be a useful first step.

Reply:
@tomdyson I'd say you're right - that looks like exactly the same thing I'd implemented for a test project to host multiple sites on wagtail with templates based upon which site was being served. I don't like the hacked global using thread local storage though; is there some way that the site could be passed through better to the template loader?

Reply:
This thread feels like it's diverged from the actual PR quite a bit - is there a rephrasing that can be applied to this paragraph so that this can land and wagtail users have docs to fall back on when they need to set multiple 404s? =)

Reply:
I suggest updating line 13 to the following:

> Note that users with access to the admin can log in through every hostname that has been defined for a site. After logging in, however, all user rights that were assigned via groups are honoured.